python/paddle/utils/merge_model.py

```diff
@@ -70,3 +113,164 @@ def merge_v2_model(net, param_file, output_file):
+                            (param_value[index] - param_min) / bucket_width)
+                        param_value[index] = np.float32((
+                            bucket + 0.5) * bucket_width + param_min)
+                param_value = param_value.reshape(param_shape)
+            else:
+                # print ("We retain the original %s because it's size is below 8192" % param_name)
+                pass
+        return self.param
+
+
+class Merge_BN:
+    def __init__(self, source_proto, source_param):
+        self.source_proto = source_proto
+        self.parameters = self.source_proto.parameters
+        self.param_names = [param.name for param in self.parameters]
+
+        self.layers = self.source_proto.layers
+        self.sub_models = self.source_proto.sub_models[0]
+        self.input_layers_name = {}
+        self.delete_param_name = []
+        self.delete_layer_name = []
+
+        self.source_param = source_param
+
+    def fuse_param(self, current_layer, bn_layer):
+        current_layer.active_type = bn_layer.active_type
+
+        param_name = current_layer.inputs[0].input_parameter_name
+        bias_name = current_layer.bias_parameter_name
+        assert param_name, 'This layer(fc or exconv) should have parameters'
+
+        bn_inputs = bn_layer.inputs
+
+        a_bn_name = bn_inputs[0].input_parameter_name
+        mean_bn_name = bn_inputs[1].input_parameter_name
+        var_bn_name = bn_inputs[2].input_parameter_name
+        b_bn_name = bn_layer.bias_parameter_name
+
+        a_bn = self.source_param.get(a_bn_name)
+        mean_bn = self.source_param.get(mean_bn_name)
+        var_bn = self.source_param.get(var_bn_name)
+        b_bn = self.source_param.get(b_bn_name)
+
+        param = self.source_param.get(param_name)
+        param_shape = param.shape
+
+        bias = np.zeros(a_bn.shape[1])
+        bias_shape = (bias.size, 1)
+
+        # the parameters in batch normalization will be unuseful
+        self.delete_param_name.append(a_bn_name)
+        self.delete_param_name.append(mean_bn_name)
+        self.delete_param_name.append(var_bn_name)
+
+        if not bias_name:
+            bias_name = param_name.split('.')[0] + '.wbias'
+            current_layer.bias_parameter_name = bias_name
+            # 1.add conv bias parameter config to prototxt
+            # 2.add conv bias parameters config to prameters
+            bias_shape = b_bn.shape
+            bias_param_config = self.parameters[self.param_names.index(
+                b_bn_name)]
+            bias_param_config.name = bias_name
+            self.source_param.__append_config__(bias_param_config)
+        else:
+            self.delete_param_name.append(b_bn_name)
+            bias = self.source_param.get(bias_name).reshape(1, -1)
+
+        std_bn = np.float32(np.sqrt(np.add(var_bn, 1e-5)))
+        tmp1 = np.float32(np.divide(a_bn, std_bn))
+        bias = np.float32(
+            np.add(np.multiply(np.subtract(bias, mean_bn), tmp1),
```

Comment:
I think the bias is not need here. Of course, bias of current layer is usually not present, so it works well when I tried this with MobileNet.

Reply:
You are right. There exactly does not need to deal with the `bias` in the normal case.  Adding this just to avoid people who use batch norm but also add bias after conv.

Reply:
What I mean is that we can ignore the bias even if adding bias after conv, but I realized I have made a mistake, so the code is right. 

Thanks~